libyul/AsmParser.cpp

```diff
@@ -327,6 +327,9 @@ variant<Literal, Identifier> Parser::parseLiteralOrIdentifier()
 	case Token::HexStringLiteral:
 		fatalParserError(3772_error, "Hex literals are not valid in this context.");
 		break;
+	case Token::Illegal:
+		fatalParserError(1465_error, "Illegal token: " + to_string(m_scanner->currentError()));
+		break;
```

Comment:
Would it make sense to have different error codes for different scanner errors? Or would that be an overkill?

Reply:
I'd say overkill for this release. We could rethink error reporting of parser/scanner errors in general, and make it uniform across solidity/yul.

Reply:
ok. I asked about this particular case mostly because the error messages seem to vary a lot (even though they're all ultimately about invalid tokens).